src/services/pagoPAClientFactory.ts

```diff
@@ -2,17 +2,27 @@
  * This service builds API clients.
  */
 
-import { IPagoPAClientFactoryInterface } from "./IPagoPAClientFactory";
+import {
+  IPagoPAClientFactoryInterface,
+  PagoPAEnvironment
+} from "./IPagoPAClientFactory";
 
 import { PagoPAClient } from "../clients/pagopa";
 
 export default class PagoPAClientFactory
   implements IPagoPAClientFactoryInterface {
-  constructor(public readonly pagoPAApiUrl?: string) {}
+  constructor(
+    public readonly pagoPAApiUrl?: string,
```

Comment:
these should be both required

Reply:
This change needs a refactoring of tests where is present the line 
```const pagoPAClientFactory = new PagoPAClientFactory();```
How can i refactor those lines? This works but i don't know if it is correct
```
const pagoPAClientFactory = new PagoPAClientFactory(
    process.env.PAGOPA_API_URL as string,
    process.env.PAGOPA_API_URL_TEST as string
);
```

Reply:
casting is seldom the right thing to do, in this case we don't care much since it happens during tests, if something breaks we get alerted - so no big deal

Reply:
or else you can just pass strings here during tests, instead of reading them from the env

Reply:
the point here is that we can't have a `PagoPAClientFactory` without both `pagoPAApiUrl` or `pagoPAApiUrlTest` being defined